trieste/models/keras/models.py

```diff
@@ -249,7 +250,11 @@ def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
         :return: The predicted mean and variance of the observations at the specified
             ``query_points``.
         """
-        ensemble_distributions = self.ensemble_distributions(query_points)
+        # handle leading batch dimensions, while still allowing `Functional` to
+        # "allow (None,) and (None, 1) Tensors to be passed interchangeably"
+        input_dims = min(len(query_points.shape), len(self.model.input_shape[0]))
```

Comment:
Very minor point, `tf.rank(query_points)` would be a bit more concise.

Reply:
Actually that confused AutoGraph for some reason in one of the integration tests! I'm somewhat pressed for time at the moment so might leave as is for now and try and figure out why later.

Reply:
I have a hunch why that may be ... using `tf.rank` would create a tensor and the autograph would try to integrate that into the graph. In my experience adding `.numpy()` at the end, or some other way of making it static usually works. Not worth the hassle in this case though ;-)